### PR TITLE
NPS comments

### PR DESCRIPTION
**Pull request by Lukewh**
Fixes https://github.com/canonical-websites/snapcraft.io/issues/1385

## QA
- Pull the branch
- `./run`
- Visit http://0.0.0.0:8004/?nps=true
- Open the network tab of the dev console
- Click "Provide further comments" in the notification
- Add some text
- Click Send
- The button should show a spinner, followed by a Thanks message.
- Check the network tab of the dev console to see the POST to marketo

### CODE REVIEW

**Comment by therealjuan** [comment]
Can we dismiss the notification once we enter the comment?

**Comment by Lukewh** [comment]
Done @therealjuan

**Comment by bartaz** [comment]
I feel like having "Dismiss" button in exactly same place and same style as "Send" button is a bit misleading.
I clicked "Send" looked at network panel to confirm it worked, looked back at the form and I thought that the form didn't change the state. I didn't want to click "Dismiss" button because it felt like I would be sending the same information again.

I think we need some more feedback to user. Maybe hiding textarea after sending the data?

screen shot 2018-12-17 at 09 32 48

**Comment by therealjuan** [comment]
I think that closing down the textarea will help to convey the message that things have changed. So maybe that would be enough?

**Comment by Lukewh** [comment]
@bartaz @therealjuan updated xox

**Comment by therealjuan** [comment]
Ace!

File: templates/index.html [hunk 1]
```diff
@@ -7,10 +7,19 @@
...
-        <p class="p-notification__response">
-          Thank you so much 🙏. Your feedback is really valuable to us.
-        </p>
+      <div class="p-notification p-snapcraft-nps-banner js-notification-holder">
+        <div class="p-notification__response">
+          <p class="u-no-margin--bottom u-no-padding--top">
+            Thank you so much 🙏. Your feedback is really valuable to us. <a href="#" class="js-nps-comment-toggle">Provide further comments</a>
+          </p>
+          <div class="p-snapcraft-nps-banner__form js-nps-comment u-hide">
+            <form class="--dark">
```
Comment (bartaz) [inline templates/index.html]: Why being so `--dark`?

File: templates/index.html [hunk 2]
```diff
@@ -7,10 +7,19 @@
...
+              <textarea rows="5" name="snapcraftNPSNotes"></textarea>
+              <button type="submit" class="has-icon u-float--right">Send</button>
+            </form>
+          </div>
+        </div>
+        <a href="#" class="p-notification__action"><i class="p-icon--close"></i></a>
```
Comment (bartaz) [inline templates/index.html]: Why not using vanilla built in notification support for `<button class="p-icon--close">` without custom "__action"?

```html
<div class="p-notification" id="notification">
  <p class="p-notification__response">
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nesciunt ipsum nemo autem reiciendis nulla tempore natus repudiandae dolorem. Corporis maxime, iure maiores repellat, odit facilis!
  </p>
  <button class="p-icon--close" aria-label="Close notification" onclick="closeNotification('notification')">Close</button>
</div>
```

File: templates/index.html (hunk 1 above)
Comment (Lukewh) [reply templates/index.html]: So the icon in the button is the light version... The suggested use is in a `p-strip--dark` but I don't want a script!

File: templates/index.html (hunk 2 above)
Comment (Lukewh) [reply templates/index.html]: A thanks, will update this - I couldn't find it in the docs so thought I'd imagined it :+1:!

File: templates/index.html (hunk 1 above)
Comment (bartaz) [reply templates/index.html]: Ah, the spinner. Would it work with the class on the button? And a comment in the template please, so we remember why it's there.

File: templates/index.html [hunk 3]
```diff
@@ -7,10 +7,19 @@
...
+            Thank you so much 🙏. Your feedback is really valuable to us. <a href="#" class="js-nps-comment-toggle">Provide further comments</a>
+          </p>
+          <div class="p-snapcraft-nps-banner__form js-nps-comment u-hide">
+            <form>
+              <textarea rows="5" name="snapcraftNPSNotes"></textarea>
+              <button type="submit" class="has-icon u-float--right button--dark">Send</button>
+            </form>
+          </div>
+        </div>
+        <a arial-label="Close notification" class="p-icon--close p-notification__close"><i class="p-icon--close"></i></a>
```
[the remark this answers is not in the document]
Comment (Lukewh) [reply templates/index.html]: If it's a button it has a border and the p-icon--close doesn't do anything. Is this the wrong version of Vanilla?

[the remark this answers is not in the document]
Comment (Lukewh) [reply templates/index.html]: Strange, I just changed it to take a screenshot and it worked :man_shrugging:

**Pull request merged**